src/core/basic_client.rs

```diff
@@ -24,16 +24,23 @@ use std::collections::HashSet;
 /// The client exposes low-level functionality for using the Parsec service.
 /// Below you can see code examples for a few of the operations supported.
 ///
+/// For all cryptographic operations an implicit provider is used which can be
+/// changed between operations. The client starts with no such defined provider
+/// and it is the responsibility of the user to identify and set an appropriate
+/// one. As such, it is critical that before attempting to use cryptographic
+/// operations users call [`list_providers`](#method.list_providers)
+/// and [`list_provider_operations`](#method.list_provider_operations)
```

Comment:
Not totally related with this PR but is there a reason why using `list_provider_operations` instead of `list_opcodes`?

Reply:
I think that sounded more developer-friendly to me at the time, but maybe the low-level one should be named consistently - I'll change in another PR

Reply:
Thanks! It think so too.